typings/index.d.ts

```diff
@@ -431,6 +431,54 @@ export class CommandInteraction extends Interaction {
   private _createOptionsCollection(options: unknown, resolved: unknown): Collection<string, CommandInteractionOption>;
 }
 
+export class CommandInteractionOptionResolver {
+  public constructor(client: Client, options: CommandInteractionOption[]);
+  public readonly client: Client;
+  private _options: CommandInteractionOption[];
+  private _group: string | null;
+  private _subCommand: string | null;
+  private _getTypedOption(
+    name: string,
+    types: ApplicationCommandOptionType[],
+    properties: (keyof ApplicationCommandOption)[],
+    required: true,
+  ): CommandInteractionOption;
+  private _getTypedOption(
+    name: string,
+    types: ApplicationCommandOptionType[],
+    properties: (keyof ApplicationCommandOption)[],
+    required: boolean,
+  ): CommandInteractionOption | null;
+
+  public get(name: string, required: true): CommandInteractionOption;
+  public get(name: string, required?: boolean): CommandInteractionOption | null;
+  public getSubCommand(): string;
```

Comment:
Shouldn't this be the same as the functions below?

Reply:
No, because https://github.com/discordjs/discord.js/pull/6107/commits/6d4f5ad11bc8c59321c946e87af8f74ec5beca90#diff-b7a378d6e6f1796a78cb4f3d3465eb376681eb389a2980c4b7c955642b213beaR91 -- according to the API docs, you can't have any other option types at the same level as a sub-command/sub-command group, so it makes sense to assert this.